src/upload.ts

```diff
@@ -70,10 +76,9 @@ export class S3Uploader {
       // Make sure we re-initialize the connection next time
       this.s3 = undefined;
       if (retry) {
-        this.upload({ bundle, bundleId, timestamp, referrer });
-      } else {
-        throw error;
+        return this.upload({ bundle, bundleId, timestamp, referrer });
```

Comment:
This should be awaited and not just returned, right?
If we only return the promise here it's still not fully awaited (for that the caller would have to double `await` the `upload` function - once because the function is `async` and once because it returns another promise).
```suggestion
        await this.upload({ bundle, bundleId, timestamp, referrer });
```